File: Notification-Service/src/test/java/byrd/riley/notificationservice/test/model/EmbeddedAMQPBroker.java
package byrd.riley.notificationservice.test.model;

import java.net.URL;
import java.util.Map;
import java.util.Objects;

import org.apache.qpid.server.SystemLauncher;
import org.junit.jupiter.api.extension.AfterAllCallback;
import org.junit.jupiter.api.extension.BeforeAllCallback;
import org.junit.jupiter.api.extension.ExtensionContext;

public class EmbeddedAMQPBroker implements BeforeAllCallback, AfterAllCallback {
	private static final String CONFIG_FILE_NAME = "qpid-config.json";
	private static final URL INITIAL_CONFIGURATION_LOCATION = EmbeddedAMQPBroker.class.getClassLoader().getResource(CONFIG_FILE_NAME);
	private static final int BROKER_PORT = 5672;
	
	private final SystemLauncher broker = new SystemLauncher();


	@Override
	public void beforeAll(ExtensionContext context) throws Exception {
		Objects.requireNonNull(INITIAL_CONFIGURATION_LOCATION);

		Map<String, Object> attributes = Map.of(
			"type", "Memory",
			"initialConfigurationLocation", INITIAL_CONFIGURATION_LOCATION.toExternalForm(),
			"qpid.amqp_port", String.valueOf(BROKER_PORT)
		);

		broker.startup(attributes);
	}

	@Override
	public void afterAll(ExtensionContext context) { broker.shutdown(); }
}

File: Notification-Service-Kotlin/src/test/kotlin/byrd/riley/notificationservicekotlin/test/model/EmbeddedAMQPBroker.kt
package byrd.riley.notificationservicekotlin.test.model

import org.apache.qpid.server.SystemLauncher
import org.junit.jupiter.api.extension.AfterAllCallback
import org.junit.jupiter.api.extension.BeforeAllCallback
import org.junit.jupiter.api.extension.ExtensionContext

class EmbeddedAMQPBroker: BeforeAllCallback, AfterAllCallback {
    companion object {
        private const val CONFIG_FILE_NAME = "qpid-config.json"
        private const val BROKER_PORT = 5672
        private val INITIAL_CONFIGURATION_LOCATION = Companion::class.java.classLoader.getResource(CONFIG_FILE_NAME)
    }

    private val broker = SystemLauncher()

    override fun beforeAll(context: ExtensionContext?) {
        val attributes = mapOf<String, Any>(
            "type" to "Memory",
            "initialConfigurationLocation" to INITIAL_CONFIGURATION_LOCATION!!.toExternalForm(),
            "qpid.amqp_port" to BROKER_PORT.toString()
        )

        broker.startup(attributes)
    }

    override fun afterAll(context: ExtensionContext?) {
        broker.shutdown()
    }
}